mapie/quantile_regression.py

```diff
@@ -0,0 +1,486 @@
+            Calibration labels.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+
+            By default ``None``.
+
+        Returns
+        -------
+        MapieQuantileRegressor
+             The model itself.
+        """
+        # Checks
+        self._check_parameters()
+        checked_estimator = self._check_estimator(self.estimator)
+        alpha = self._check_alpha(self.alpha)
+        self.cv = self._check_cv(cast(str, self.cv))
+        X, y = indexable(X, y)
+        X_calib, y_calib = indexable(X_calib, y_calib)
+        y = _check_y(y)
+        y_calib = _check_y(y_calib)
+        self.n_samples_calib = _num_samples(y_calib)
+        check_alpha_and_n_samples(self.alpha, self.n_samples_calib)
+        self.n_features_in_ = check_n_features_in(
+            X,
+            estimator=checked_estimator
+            )
+        sample_weight, X, y = check_null_weight(
+            sample_weight,
+            X,
+            y
+        )
+        y = cast(NDArray, y)
+
+        # Initialization
+        self.estimators_: List[RegressorMixin] = []
+
+        # Work
+        y_calib_preds = np.full(
+            shape=(3, self.n_samples_calib),
+            fill_value=np.nan
+        )
+
+        if isinstance(checked_estimator, Pipeline):
+            estimator = checked_estimator[-1]
+        else:
+            estimator = checked_estimator
+        name_estimator = estimator.__class__.__name__
+        alpha_name = self.quantile_estimator_params[
+            name_estimator]["alpha_name"]
+        for i, alpha_ in enumerate(alpha):
+            cloned_estimator_ = clone(checked_estimator)
+            params = {alpha_name: alpha_}
+            if isinstance(checked_estimator, Pipeline):
+                cloned_estimator_[-1].set_params(**params)
+            else:
+                cloned_estimator_.set_params(**params)
+            self.estimators_.append(fit_estimator(
+                cloned_estimator_, X, y, sample_weight
+            ))
+            y_calib_preds[i] = self.estimators_[-1].predict(X_calib)
+
+        self.conformity_scores_ = np.full(
+                shape=(3, self.n_samples_calib),
+                fill_value=np.nan
+            )
+        self.conformity_scores_[0] = y_calib_preds[0]-y_calib
+        self.conformity_scores_[1] = y_calib-y_calib_preds[1]
+        self.conformity_scores_[2] = np.max(
+            [
+                self.conformity_scores_[0],
+                self.conformity_scores_[1]
+            ], axis=0
+        )
+        return self
+
+    def predict(  # type: ignore
```

Comment:
Still needed ?

Reply:
Either we put the format:
`def predict(
        self,
        X: ArrayLike,
        ensemble: Optional[bool] = False,
        alpha: Optional[Union[float, Iterable[float]]] = None,
        symmetry: Optional[bool] = True,
    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:` 
where we do not use ensemble or alpha or we use this "cleaner" format but make a `# type: ignore`

Otherwise, this error would occur:
"""
mapie/quantile_regression.py:406: error: Signature of "predict" incompatible with supertype "MapieRegressor"
mapie/quantile_regression.py:406: note:      Superclass:
mapie/quantile_regression.py:406: note:          def predict(self, X: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], ensemble: bool = ..., alpha: Union[float, Iterable[float], None] = ...) -> Union[ndarray[Any, dtype[Any]], Tuple[ndarray[Any, dtype[Any]], ndarray[Any, dtype[Any]]]]
mapie/quantile_regression.py:406: note:      Subclass:
mapie/quantile_regression.py:406: note:          def predict(self, X: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], symmetry: Optional[bool] = ...) -> Union[ndarray[Any, dtype[Any]], Tuple[ndarray[Any, dtype[Any]], ndarray[Any, dtype[Any]]]]
"""

Reply:
Right I forgot this point, it would be worth discussing with Thomas on this problem, he might have some valuable insight.

Reply:
Yes indeed. He might have the key to this issue.